blobstore-uploader/src/main/java/org/jclouds/examples/blobstore/BlobUploader.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jclouds.examples.blobstore;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.Properties;
+import java.util.UUID;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.FileUtils;
+import org.jclouds.Constants;
+import org.jclouds.ContextBuilder;
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.domain.Blob;
+import org.jclouds.domain.Location;
+import org.jclouds.io.Payloads;
+import org.jclouds.openstack.swift.v1.blobstore.RegionScopedBlobStoreContext;
+
+public class BlobUploader implements Runnable {
+   /**
+    * ThreadLocal allows us to use 1 container and 1 connection per thread and reuse them.
+    * They need to be static, but here we also have to instantiate them at runtime, as we pass parameters from
+    * the command line.
+    */
+   private static ThreadLocal<BlobStore> blobStore = new ThreadLocal<BlobStore>();
+   private static ThreadLocal<String> container = new ThreadLocal<String>();
+
+   /**
+    * The only thing that really needs to be passed to this unit of work is the File to be uploaded.
+    * The credentials are here for convenience.
+    */
+   private String username;
+   private String password;
+   private String provider;
+   private String region;
+   private File file;
+
+   public
+   BlobUploader(String username, String password, String provider, String region, File file) {
+      this.username = username;
+      this.password = password;
+      this.provider = provider;
+      this.region = region;
+      this.file = file;
+   }
+
+   @Override
+   public void run() {
+      /**
+       * Instantiate the ThreadLocal variables when this thread runs for the first time.
+       * Instantiating this in the constructor will not work (different thread).
+       */
+      if(blobStore.get() == null) {
+         // It is usually a good idea to include the currentThread when logging parallel tasks.
+         System.out.println("Creating connection for thread " + Thread.currentThread());
+         /**
+          * In some cases, especially when running very large jobs with many parallel threads, some connections will
+          * break. In that case, we need to be able to obtain a new connection (and socket) to the service, which is
+          * why this is factored out.
+          */
+         resetBlobstore(username, password, provider, region);
+      }
+
+      if(container.get() == null) {
+         container.set(UUID.randomUUID().toString());

```

Comment:
I should probably add some comments to clarify why UUID.
You want a randomized container name (for aws). You also want a container per thread (makes this much simpler to use with other providers and in code). Does this make sense?